### PR TITLE
IGAPP-136: Internal linking

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

This PR generalizes internal linking and navigating by using the types and pathname parser from #134.
It also generalizes the way news types are used.

### CODE REVIEW

**Comment by klinzo** [comment]
Testable at testumgebung > Test mit Links

File: native/src/routes/news/containers/NewsContainer.js [hunk 1]
```diff
@@ -75,15 +75,17 @@ const onRouteClose = (routeKey: string, dispatch: Dispatch<StoreActionType>) =>
...
-    forceRefresh: true,
-    key: route.key
-  })
+    newsType: selectedNewsType,
+    languageCode: language,
+    newsId: newsId || undefined
+  },
+  route.key,
+  true
+  )
```
Comment (maxammann) [inline native/src/routes/news/containers/NewsContainer.js:88]: The formatting looks off for me here. I hink the `route.key` and `true` should be intended

Comment (maxammann) [reply native/src/routes/news/containers/NewsContainer.js:88]:
```
  navigateTo({
    route: POIS_ROUTE,
    cityCode,
    languageCode: language,
    cityContentPath: path || undefined
  }, route.key, true)
```

This formatting is used elsewhere in this PR.

**Review by maxammann — APPROVED** [review]

**Comment by klinzo** [comment]
Most of our routes do not support linking to other cities or languages (except from categories, events and pois). I asked @dkehne and he said cities never (intentional) link to other cities and languages though. Should we handle this and prevent this somehow or do you think it is just not supported and crashes/wrong behavior is acceptable if it is done anyway?

Also: I did encounter some crashes when navigating to the search, but were not able to reproduce it the last times I tried.

File: native/package.json [hunk 2]
```diff
@@ -82,7 +82,7 @@
     "react-native-static-server": "^0.5.0",
     "react-native-svg": "^12.1.0",
     "react-native-vector-icons": "^7.1.0",
-    "react-native-webview": "^10.9.1",
+    "react-native-webview": "10.8.2",
```
Comment (Znippet) [inline native/package.json:85]: Do we have a bug-task for this? Will hurt us if we try to upgrade native dependencies

**Review by Znippet — APPROVED** [review]

File: native/package.json (hunk 2 above)
Comment (klinzo) [reply native/package.json:85]: Will create one :+1:

**Pull request merged**